Add common Java programming words to spelling & heading exceptions

michelle-purcell opened the pull request:
The following names/services/components are regularly used in Java programming documentation in the Software industry but flag as spelling or capitalization errors by the Red Hat Vale rules. 

- classloader & ClassLoader
- classpath
- ~~foobar~~
- GraalVM
- Hibernate ORM
- JUnit
- Kafka
- Mandrel
- MicroProfile
- MongoDB
- mutual TLS (and initialism: mTLS)
- OpenID Connect
- WebAuthn

This PR adds those proper nouns to the spelling, heading, and capitalization style rules.

**Fixes:**
Lots of false positives, particularly in Runtimes product docs, for example, **Red Hat build of Quarkus**.

apupier commented on .vale/styles/RedHat/Spelling.yml:
```diff
@@ -151,6 +151,8 @@ filters:
   - Che-Theia
   - Ciphertext
   - Civetweb
+  - ClassLoader
```
For this one, I just want to mention that in Java world we are seeing various notations:
- ClassLoader (which initially was to refer to the classname https://docs.oracle.com/javase/7/docs/api/java/lang/ClassLoader.html )
- class loader or Class Loader (with a space, maybe the most used notation in Java specification, 59 times on the [page](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-5.html) describing it)
- classloader or Classloader (without space, for instance used in red hat doc [here](https://access.redhat.com/documentation/en-us/jboss_enterprise_application_platform/5/html/microcontainer_user_guide/sect-jboss_microcontainer_user_guide-using_services-dynamic_classloading) and in Java specification [here](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-5.html#jvms-5.3) and in [title of wikipedia article](https://en.wikipedia.org/wiki/Java_Classloader))

michelle-purcell replied:
@apupier - Thanks for flagging this. Wondering if we need to add exceptions for all of the above variations. What do you think?

apupier replied:
Given how much the different variations are spread, I think it is more consensual to allow all of these variations. So I vote for allowing all of them. But I'm not a Technical Writer so take my words with a grain of salt.

michelle-purcell replied:
I changed the spelling yml instance to _classloader_ to prevent a spelling error on any of these: classloader/Classloader/ClassLoader.

As _ClassLoader_ is correct, when used to refer specifically to the Java class name and _classloader_ is also a correct form, I added ClassLoader to the Headings.yml.  
The other variation, _class loader_ doesn't generate a vale warning.

If there's any further discussion needed, we can bring this to the CCS style council for advice.

Thanks @apupier :+1:

rolfedh commented on .vale/fixtures/RedHat/CaseSensitiveTerms/testvalid.adoc:
```diff
@@ -20,6 +21,7 @@ Exec-Shield
 Exif
 FAQ
 Fedora™ Project
+foobar
```
ISG:
fubar noun (also "foobar" and derivatives)
Do not use. This term is an acronym of a profanity that is sometimes used by developers in code. If you see this term in sample code, revise the example or do not publish it. Rewrite any text to omit the term.

michelle-purcell replied:
@rolfe - Thanks for catching this. I searched IBM style for the string 'foobar', which doesn't come up so I ploughed on. I see it now under fubar and foo.

michelle-purcell replied:
Fixed.

rolfedh commented on .vale/fixtures/RedHat/Spelling/testinvalid.adoc:
```diff
@@ -8,14 +8,18 @@ btrfs
 dns
 dotnet
 endevor
+foo bar
```
See previous comment.

michelle-purcell replied:
Fixed.

rolfedh commented on .vale/fixtures/RedHat/Spelling/testvalid.adoc:
```diff
@@ -96,13 +98,15 @@ failback
 Failback
 failover
 Failover
+foobar
```
See prev comment.

michelle-purcell replied:
Fixed

rolfedh commented on .vale/fixtures/RedHat/Spelling/testvalid.adoc:
```diff
@@ -170,6 +177,7 @@ Mirantis
 Mixin
 Mixins
 Modularization
+Mongo DB
```
It [looks like](https://www.google.com/search?q=Mongo+DB&sourceid=chrome&ie=UTF-8) "MongoDB" is the official spelling.

michelle-purcell replied:
Fixed.

rolfedh commented on .vale/styles/RedHat/Headings.yml:
```diff
@@ -105,11 +108,14 @@ exceptions:
   - Minishift
   - Mirantis
+  - MongoDB
+  - Mutual TLS
```
Also known as `mTLS`. Would it make sense to add that, too?

michelle-purcell replied:
Done. Added mTLS to the definitions.yml file. :+1:

michelle-purcell commented:
@apupier & @rolfedh - Thank you for your review and feedback. I have fixed all of the action items that you suggested. 

I also added Hibernate ORM and Mandrel.
